pkg/diagnostics/errorcode_monitoring.go

```diff
@@ -0,0 +1,52 @@
+package diagnostics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"github.com/dapr/dapr/pkg/messages/errorcodes"
+)
+
+type errorCodeMetrics struct {
+	errorCodeCount *stats.Int64Measure
+
+	appID   string
+	ctx     context.Context
+	enabled bool
+}
+
+func newErrorCodeMetrics() *errorCodeMetrics {
+	return &errorCodeMetrics{ //nolint:exhaustruct
+		errorCodeCount: stats.Int64(
+			"error_code/count",
+			"Number of times an error with a specific errorcode was encountered.",
+			stats.UnitDimensionless),
+
+		ctx:     context.Background(),
+		enabled: false,
+	}
+}
+
+// Init registers the errorcode metrics view.
+func (m *errorCodeMetrics) Init(id string) error {
+	m.enabled = true
+	m.appID = id
+
+	return view.Register(
+		diagUtils.NewMeasureView(m.errorCodeCount, []tag.Key{appIDKey, errorCodeKey, categoryKey}, view.Count()),
+	)
+}
```

Comment:
Remove, and move to `New`.

Reply:
the metric is optional and so the Init() is a secondary function to call when the flag is checked

https://github.com/jake-engelberg/dapr//blob/5c3b42bf03519008739c1fc2ec8030dba5f1fd00/pkg/diagnostics/metrics.go#L53